dm/pkg/checker/table_structure.go

```diff
@@ -149,41 +144,78 @@ func (c *TablesChecker) Name() string {
 	return "table structure compatibility check"
 }
 
-func (c *TablesChecker) checkCreateSQL(ctx context.Context, statement string) []*incompatibilityOption {
-	parser2, err := dbutil.GetParserForDB(ctx, c.db)
+func (c *TablesChecker) checkTable(ctx context.Context, r *Result) {
+	defer c.checkWg.Done()
+	sourceID := c.firstSourceID
+	p, err := dbutil.GetParserForDB(ctx, c.dbs[sourceID])
 	if err != nil {
-		return []*incompatibilityOption{
-			{
-				state:      StateFailure,
-				errMessage: err.Error(),
-			},
-		}
+		c.errCh <- err
+		c.cancel()
+		return
```

Comment:
So you want to cancel all check progress if any error happended? If so, I think [errGroup](https://pkg.go.dev/golang.org/x/sync/errgroup) meets your requriment. You can just return when error occur.